pkg/control/v1/client/dial.go

```diff
@@ -14,18 +14,44 @@ import (
 	"github.com/elastic/elastic-agent/pkg/control"
 
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
 	"google.golang.org/grpc/credentials/insecure"
 )
 
 func dialContext(ctx context.Context) (*grpc.ClientConn, error) {
-	return grpc.DialContext( //nolint:staticcheck // Only the deprecated version allows this call to be blocking
+	return dialContextBlocking(
 		ctx,
 		strings.TrimPrefix(control.Address(), "unix://"),
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithContextDialer(dialer),
 	)
 }
 
+// dialContextBlocking creates a blocking connection equivalent to the deprecated grpc.DialContext
+func dialContextBlocking(ctx context.Context, target string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
It looks like you have this function defined 4 times in different packages. You should move it to a common location, for example `pkg/control/dial.go`.

In addition, we should have at least one unit test for it.

Reply:
yeah makes sense @swiatekm i moved it to pkg/control/dial.go and added a basic unit test for now
let me know if you think it needs more coverage or changes